### PR TITLE
Api entity validations MaterialItem

### PR DESCRIPTION
**Pull request by BacLuc**

### CODE REVIEW

File: api/src/Entity/MaterialItem.php [hunk 1]
```diff
@@ -75,6 +77,10 @@ class MaterialItem extends BaseEntity implements BelongsToCampInterface, CopyFro
     /**
      * The name of the item that is required.
      */
+    #[InputFilter\Trim]
+    #[InputFilter\CleanText]
+    #[Assert\NotBlank]
+    #[Assert\Length(max: 32)]
```
Comment (usu) [inline api/src/Entity/MaterialItem.php]: Isn't this is a bit too restrictive, to allow only 32 chars for the article name?

**Review by usu — APPROVED** [review]

File: api/src/Entity/MaterialItem.php (hunk 1 above)
Comment (BacLuc) [reply api/src/Entity/MaterialItem.php]: What is your suggestion? 64, 128?
On mobile, 32 is a lot.
image
There we need a wrapping text input field that the user can edit the article.
On Desktop it is not a problem.
On PDF we can handle long texts.
It does not look nice on the pdf, but the user sees the information.

I would vote for 64, this was enough for my camps in ecamp2.

Comment (usu) [reply api/src/Entity/MaterialItem.php]: 64 sounds good to me.

File: api/tests/Api/MaterialItems/UpdateMaterialItemTest.php [hunk 2]
```diff
@@ -304,6 +359,43 @@ public function testPatchMaterialItemValidatesInvalidQuantity() {
...
+            'quantity' => 1.7E308,
+        ], 'headers' => ['Content-Type' => 'application/merge-patch+json']]);
+
+        $this->assertResponseStatusCodeSame(200);
+        $this->assertJsonContains([
+            'quantity' => 1.7E308,
+        ]);
+    }
+
+    public function testPatchMaterialItemDoesNotCrashForLargeNumberForQuantity() {
```
Comment (carlobeltrame) [inline api/tests/Api/MaterialItems/UpdateMaterialItemTest.php:376]: Nice one!

**Review by carlobeltrame — APPROVED** [review]

**Pull request merged**